### ISSUE TITLE
Error in passing weights to custom objective function LGBMClassifier

### ISSUE THREAD

**Issue by Sidshroff** [microsoft/LightGBM#6186]
## Description
I am not able to pass weights to a custom objective function with LGBM Classifier. A debugger on logloss function always shows weights as None

## Reproducible example
    def logloss(y_true, y_pred, weight):
       y_pred = 1. / (1. + np.exp(-y_pred))
       grad = (y_pred - y_true) * weight
       hess = y_pred * (1.0 - y_pred) * weight
       return grad, hess

      model = lgbm.LGBMClassifier(**parameters.lgbm_params, objective=logloss)
      train_data = lgbm.Dataset(X_train, Y_train, weight=weights, free_raw_data=False)
      trained_model = model.fit(train_data.data, train_data.label, weight=train_data.weight)

## Environment info

LightGBM version or commit hash:

Command(s) you used to install LightGBM

```shell

```

**Comment by jmoralez** [comment]
Hey @Sidshroff, thanks for using LightGBM. If you're using the scikit-learn API then you don't need the `lgbm.Dataset` object, sklearn takes the X, y and weights arrays. Also the name of the argument is `sample_weight`, not `weight`, so you can try something like the following:
```python
import lightgbm as lgb
import numpy as np
from sklearn.datasets import make_classification

def logloss(y_true, y_pred, weight):
   y_pred = 1. / (1. + np.exp(-y_pred))
   grad = (y_pred - y_true) * weight
   hess = y_pred * (1.0 - y_pred) * weight
   return grad, hess

X, y = make_classification()
weights = np.random.rand(y.size)
model = lgb.LGBMClassifier(objective=logloss).fit(X, y, sample_weight=weights)
```

Please let us know if you have further doubts.
